### ISSUE TITLE
Running ESLint via the CLI with glob syntax leads to exit code 255 on Windows

### ISSUE THREAD

**Issue by Nimelrian** [eslint/eslint#9353]
**Tell us about your environment**

* **ESLint Version:** 4.7.2
* **Node Version:** 6.3.1
* **npm Version:** 3.10.3

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    "extends": "airbnb-base",
    "plugins": [
        "import"
    ]
};


```

</details>

**What did you do? Please include the actual source code causing the issue.**
Having a folder structure like this:
```
eslint-test
|-- .eslintrc.js
|-- package.json
|           
\-- src
     \-- index.js
```
with `src/index.js` being

```js
function foo(arg) {
  return arg * 2;
}

foo(2);
```
I ran the following in Powershell (Also happens when running via an npm script)
```powershell
PS C:\dev\js\eslint-test> .\node_modules\.bin\eslint "./src/**/*.js?(x)"
```
**What did you expect to happen?**
ESLint finishes without errors/warnings. Checking `$LASTEXITCODE` in Powershell yields 0.

**What actually happened? Please include the actual, raw output from ESLint.**
ESLint finishes without errors/warnings. Checking `$LASTEXITCODE` in Powershell yields 255.
Output:
```
PS C:\dev\js\eslint-test> .\node_modules\.bin\eslint "./src/**/*.js?(x)"

C:\dev\js\eslint-test>node  "C:\dev\js\eslint-test\node_modules\.bin\\..\eslint\bin\eslint.js" ./src/**/*.js?(x)

C:\dev\js\eslint-test>
PS C:\dev\js\eslint-test>
```

Prompt is on the last line.

Running ESLint without glob syntax, but just on `./src` also shows no errors/warnings, but the `$LASTEXITCODE` yields 0.

Since ESLint exits with a non-zero value, our CI/npm scripts fail.

**Comment by Nimelrian** [comment]
Nevermind, it seems like the issue stems from the unusual escaping Powershell uses. Running ESLint via
``.\node_modules\.bin\eslint "`"./src/**/*.js?(x)`""`` fixed it (Notice the backticks in front of the inner quotes)
